### ISSUE TITLE
[a11][Violation] Accessibility issues reported on RStudio UI using IBM Equal Access Accessibility Checker tool

### ISSUE THREAD

**Issue by nkn2022** [rstudio/rstudio#15320]
### System details

    RStudio Edition : RStudio Server
    RStudio Version : 2024.04.02  Build 764
    OS Version      : RHEL 9
    R Version       : R 4.3.1

### Steps to reproduce the problem

- Install Chrome extension for **IBM Equal Access Accessibility Checker** tool. Ref: https://www.ibm.com/able/toolkit/verify/automated
- Open RStudio UI on Chrome browser 
- Run the tool and we can see that the following `Violation`s are reported.

### Describe the problem in detail

I recently ran the **IBM Equal Access Accessibility Checker** tool on the RStudio UI on a Chrome browser. The results look good overall, only a few things are missing to reach a first formal level of compliance. So kindly help look into these. 

Issues reported were violating **WCAG 2.2** requirements. 
Ref: https://www.ibm.com/able/requirements/checker-rule-sets

Listing of violations categorized based on the checkpoints. 
Full report (includes information on element locations where issue were found): 
[5.1_Accessibility_Report · RStudio Server.xlsx](https://github.com/user-attachments/files/17382753/5.1_Accessibility_Report.RStudio.Server.xlsx)

### `2.4.1 Bypass Blocks`

> **Violation of Rule:** aria_application_labelled
> **Issue:** Element with "application" role does not have a label

> **Violation of Rule:** aria_banner_label_unique
> **Issue:** Multiple elements with "banner" role do not have unique labels

### `4.1.2 Name, Role, Value`

> **Violation of Rule:** input_label_exists (Each form control must have an associated label)
> **Issue:** Form control element <textarea> has no associated label

> **Violation of Rule:** element_tabbable_role_valid (A tabbable element must have a valid widget role)
> **Issue(s):**  
> The tabbable element's role 'none' is not a widget role.
> The tabbable element's role 'document' is not a widget role
> The tabbable element's role 'none' is not a widget role.

> **Violation of Rule:** table_aria_descendants (Table structure elements cannot specify an explicit 'role' within table containers)
> **Issue**: 
> An explicit ARIA 'role' is not valid for element within an ARIA role 'table' per the ARIA in HTML specification

### `2.5.8 Minimum Target Size`

> **Violation of Rule:** target_spacing_sufficient (The target must be sufficiently sized or spaced from other targets)
> **Issue(s):** 
> Undersized target "button" does not have sufficient spacing of 12 CSS pixels from another target "button"
> Undersized target "div" does not have sufficient spacing of 12 CSS pixels from another target "div"

### `2.5.3 Label in Name`

> **Violation of Rule:** label_name_visible
> **Issue:**
> Accessible name does not match or contain the visible label text

### `ARIA specification`

> **Violation of Rule:** aria_attribute_valid (ARIA attributes should be valid for the element and ARIA role to which they are assigned)
> **Issue:**
> The ARIA attributes "aria-label" are not valid for the element <div> with implicit ARIA role "generic" 

### `HTML specification`

> **Violation of Rule:** element_id_unique (Element 'id' attribute values must be unique within a document)
> **Issue(s):**
> The element has the id "gradientFill-1" that is already in use
> The element has the id "gradientFill-2" that is already in use

### `1.3.1 Info and Relationships`

> **Violation of Rule:** table_headers_exists (Data table must identify headers)
> **Issue:** 
> Table has no headers identified

### Describe the behavior you expected

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.posit.co/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by gtritchie** [comment]
Please also see our ACR reports at: https://posit.co/accessibility.

**Comment by nkn2022** [comment]
@gtritchie @ronblum Were you able to review this? Wonder if it will help to separate the issues. Let me know. Thanks.

**Comment by gtritchie** [comment]
I've conducted full audits of RStudio so we are aware of the issues and have them tracked elsewhere. So no, there would be no value in opening issues.

**Comment by nkn2022** [comment]
Okay, sounds good. What is the best way to track fixes for these issues reported here? You mentioned these are `tracked elsewhere`. So wonder if I could just refer to that and may be close this one.

**Comment by nkn2022** [comment]
Also please note that the issues are reported based on `WCAG 2.2` requirements.

**Comment by gtritchie** [comment]
The internal auditing database is not publicly accessible (no pun intended), unfortunately.

We open tracking issues for specific accessibility issues when we're slotting them into a release backlog. There are hundreds of accessibility issues in RStudio so opening them all up front isn't useful and most end up going stale and being closed before we get to them.

We are nearing end of development for the next release so can't take any new work for that but I will attempt to pull some of these into the next release. I will mention your issue from ones I open that you had noticed so you'll get notifications.

**Comment by nkn2022** [comment]
We upgraded RStudio tool to the latest and re-scanned for accessibility violations and still see Violations reported. I have updated the description with the issues for reference.

**Comment by nkn2022** [comment]
**Update**: Scanning on upgraded version. 
RStudio Server
2024.12.1 Build 563

No violations were reported in the category **`HTML specification`**

Remaining issues reported in the description of this issue persists.
